webapp/src/components/data_selectors/jira_commentvisibility_selector/jira_commentvisibility_selector.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {ReactSelectOption} from 'types/model';
+
+import BackendSelector, {Props as BackendSelectorProps} from '../backend_selector';
+
+const stripHTML = (text: string) => {
+    if (!text) {
+        return text;
+    }
+
+    const doc = new DOMParser().parseFromString(text, 'text/html');
+    return doc.body.textContent || '';
+};
+
+type Props = BackendSelectorProps & {
+    searchCommentVisibilityFields: (params: {fieldValue: string}) => (
+        Promise<{data: {groups: {items: {name: string}[]}}; error?: Error}>
+    );
+    fieldName: string;
+};
+
+const JiraCommentVisibilitySelector = (props: Props) => {
+    const {value, isMulti, instanceID, searchCommentVisibilityFields} = props;
+    const fetchInitialSelectedValues = async (): Promise<ReactSelectOption[]> =>
+        ((!value || (isMulti && !value.length)) ? [] : commentVisibilityFields(''));
```

Comment:
This is a little hard to read, partially because it's fairly terse one-liner that spans to two lines, and it also references a variable defined below, which is a bit confusing